test/src/hooks/use-form.test.js

```diff
@@ -169,6 +169,25 @@ describe('useForm hook', () => {
       expect(result.current.state.fields.meta).toHaveProperty('foo');
     });
 
+    it('should not re-register the field', () => {
+      const { result } = renderHook(() => useForm({
+        jsonSchema: { type: 'object' },
+        onSubmit: () => {}
+      }));
+
+      act(() => {
+        result.current.fieldActions.registerField('foo');
+      });
+
+      const state = result.current.state;
+
+      act(() => {
+        result.current.fieldActions.registerField('foo');
+      });
+
+      expect(result.current.state).toBe(state);
+    });
```

Comment:
What happens if:

1 - Register all fields on form.
2 - Get the initial values and set them info form.

Can you add a test with this?